generic/ipistorm.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2020 IBM
+# Author: Harish <harish@linux.vnet.ibm.com>
+#
+
+
+import os
+import time
+import platform
+
+from avocado import Test
+from avocado import skipIf
+from avocado.utils import build, cpu, genio, git, linux_modules, process
+from avocado.utils.software_manager import SoftwareManager
+
+IS_POWER_NV = 'PowerNV' in genio.read_file('/proc/cpuinfo')
+
+
+class DBLIPIStrom(Test):
+    """
+    Storm IPIs to ensure DBL interrputs/XIVE-IPSs are triggered on XIVE
+
+    :avocado: tags=ipi,power,xive
+    """
+
+    @skipIf(IS_POWER_NV, "This test is not supported on PowerNV platform")
+    def setUp(self):
+        """
+        Install necessary packages to build the linux module
+        """
+        if 'power' not in cpu.get_family():
+            self.cancel('Test Only supported on Power')
+
+        pkgs = ['gcc', 'make', 'kernel-devel']
+
+        smm = SoftwareManager()
+        for package in pkgs:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel('%s is needed for the test to be run' % package)
+
+        git.get_repo("https://github.com/antonblanchard/ipistorm",
```

Comment:
please use zip version , that help us to use `asset.fetch` library as that have more functionality

Reply:
okay